<a href="https://colab.research.google.com/github/Shivamgond2022/demo/blob/main/Assignment_3(Q3).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Q3 Assignments

In [ ]:
import pandas as pd
import numpy as np
from scipy import stats

In [ ]:
data = pd.read_csv("/content/BuyerRatio.csv")
data

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2 entries, 0 to 1
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Observed Values  2 non-null      object
 1   East             2 non-null      int64 
 2   West             2 non-null      int64 
 3   North            2 non-null      int64 
 4   South            2 non-null      int64 
dtypes: int64(4), object(1)
memory usage: 208.0+ bytes


In [ ]:
# Spliting the data between male and female to find all ratios are similer across region
Male = data.iloc[0,1:]
Female = data.iloc[1,1:]


In [ ]:
data1 = [Male,Female]
data1

[East      50
 West     142
 North    131
 South     70
 Name: 0, dtype: object, East      435
 West     1523
 North    1356
 South     750
 Name: 1, dtype: object]

In [ ]:
# using Chi-square test to find independence of variables in a contingency table.
results = stats.chi2_contingency(data1)
print(results)

(1.595945538661058, 0.6603094907091882, 3, array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))


In [ ]:
results[1]

0.6603094907091882

# Conclusion
 We are fail to reject null hypothesis as ou p_value is higher than significant value which is 0.66 > 0.05. Which means all propertis are equle. 